src/cargo/util/rustc.rs

```diff
@@ -80,6 +82,17 @@ impl Rustc {
                 verbose_version
             )
         })?;
+        let commit_hash = extract("commit-hash: ").ok().map(|hash| {
+            assert!(
+                hash.chars().all(|ch| ch.is_ascii_hexdigit()),
+                "commit hash must be a hex string"
+            );
+            assert!(
+                hash.len() == 40 || hash.len() == 64,
+                "hex string must be generated from sha1 or sha256"
+            );
```

Comment:
Should we make these `debug_assert`s?

Or should we have a general message in production asserts/panics about being an internal error and what they should collect to report the issue?